Sources/WebSocketKit/PMCE.swift

```diff
@@ -0,0 +1,753 @@
+                                          strategy: .huffmanOnly)
+
+            self.compressorBox = NIOLoopBoundBox(CompressionAlgorithm.deflate(configuration: zccConf).compressor,
+                                                 eventLoop: channel.eventLoop)
+            
+            self.decompressorBox = NIOLoopBoundBox( CompressionAlgorithm.deflate(configuration: zcdConf).decompressor,
+                                             eventLoop: channel.eventLoop)
+            
+            logger.trace("compressor \(zccConf)")
+            logger.trace("decompressor \(zcdConf)")
+        }
+        startStreams()
+    }
+    
+    ///  Starts compress-nio streams for DEFLATE support.
+    /// - returns: Void
+    public func startStreams() {
+        do {
+            try compressorBox.value?.startStream()
+        }
+        catch {
+            logger.error("error starting compressor stream : \(error)")
+        }
+        do {
+            try decompressorBox.value?.startStream()
+        }
+        catch {
+            logger.error("error starting decompressor stream : \(error)")
+        }
+    }
+    
+    ///  Stops compress-nio streams for DEFLATE support.
+    /// - returns: Void
+    public func stopStreams() {
+        do {
+            logger.debug("PMCE: stopping compressor stream...")
+            try compressorBox.value?.finishStream()
+        }
+        catch {
+            logger.error("PMCE:error finishing stream(s) : \(error)")
+        }
+        
+        do {
+            logger.debug("PMCE: stopping decompressor stream...")
+            try decompressorBox.value?.finishStream()
+        }
+        catch {
+            logger.error("PMCE:error finishing stream(s) : \(error)")
+        }
+        
+
+    }
+
+    /// websocket send calls this to compress.
+    public func compressed(_ buffer: ByteBuffer,
+                            fin: Bool = true,
+                            opCode: WebSocketOpcode = .binary) throws -> WebSocketFrame {
+        
+        guard let channel = channel else {
+            throw IOError(errnoCode: 0, reason: "PMCE: channel not configured.")
+        }
+        let notakeover = !shouldTakeOverContext()
+
+        let startSize = buffer.readableBytes
+        
+        if logging {
+            logger.trace("compressing \(startSize) bytes for \(opCode)")
+        }
+        do {
+            var mutBuffer = buffer
+
+            if !notakeover {
+                mutBuffer = unpad(buffer:buffer)
+            }
+
+            let startTime = Date()
+            
+            let compressed =
+            try mutBuffer.compressStream(with: compressorBox.value!,
+                                         flush: .sync,
+                                         allocator: channel.allocator)
+                                        //rfc says to strip padding word off compressed
+            mutBuffer = unpad(buffer: mutBuffer)
```

Comment:
Since you are no longer working with `mutBuffer` after this point, this has no effect and just wastes cycles.
```suggestion
```